src/docfx/build/dependency/DependencyResolver.cs

```diff
@@ -115,7 +115,25 @@ internal class DependencyResolver
             queries.Remove("displayProperty");
 
             // need to url decode uid from input content
-            var (error, resolvedHref, display, xrefSpec) = _xrefMap.Value.Resolve(Uri.UnescapeDataString(uid), href, displayProperty, text, declaringFile);
+            var (error, resolvedHref, xrefSpec) = _xrefMap.Value.Resolve(Uri.UnescapeDataString(uid), href);
+
+            var name = xrefSpec?.GetXrefPropertyValue("name");
+
+            // for internal UID, the display property can not be markdown or inline markdown
+            // because the display text should be plain text
+            string displayPropertyValue = null;
+            if (xrefSpec is InternalXrefSpec internalSpec
+                && !new JsonSchemaContentType[] { JsonSchemaContentType.Markdown, JsonSchemaContentType.InlineMarkdown }
+                .Contains(internalSpec.GetXrefPropertyContentType(displayProperty)))
+            {
+                displayPropertyValue = xrefSpec.GetXrefPropertyValue(displayProperty);
+            }
+
+            // fallback order:
+            // text -> xrefSpec.displayPropertyName -> xrefSpec.name -> uid
```

Comment:
any property in this fallback chain could in theory have `contentType == markdown`, so implement the filter in `GetXrefPropertyValue`? consider name `GetXrefPropertyAsString` to indicate there is a string conversion and properties might be dropped if the not a literal.

`GetXrefPropertyAsString` could return null for empty strings so we can use `??` here for fallback